src/Microsoft.Data/GroupBy.cs

```diff
@@ -0,0 +1,421 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A GroupBy class that is typically the result of a DataFrame.GroupBy call.
+    /// It holds information to perform typical aggregation ops on it.
+    /// </summary>
+    public abstract class GroupBy
+    {
+        /// <summary>
+        /// Compute the number of non-null values in each group 
+        /// </summary>
+        /// <returns></returns>
+        public abstract DataFrame Count();
+
+        /// <summary>
+        /// Return the first value in each group
+        /// </summary>
+        /// <returns></returns>
+        public abstract DataFrame First();
+
+        /// <summary>
+        /// Returns the first numberOfRows rows of each group
+        /// </summary>
+        /// <param name="numberOfRowsInEachGroup"></param>
```

Comment:
This param name doesn't match the actual parameter. You should be getting warnings for these kinds of mismatches.